cupy/cuda/compiler.py

```diff
@@ -145,6 +145,17 @@ def _get_arch():
         return min(arch, _nvrtc_max_compute_capability)
 
 
+def _get_arch_for_options(arch=None, jitify=False):
+    # This is needed to differentiate between
+    # compute or sm depending on the CUDA version
+    # needed to ensure backwards compatibility with nvrtc
+    if arch is None:
+        arch = _get_arch()
+    if not runtime.is_hip and _cuda_hip_version >= 11010 and not jitify:
```

Comment:
Let's remove the HIP check, as when this function is called we are guaranteed on CUDA:
```suggestion
    if _cuda_hip_version >= 11010 and not jitify:
```